pkg/local_object_storage/engine/gc_test.go

```diff
@@ -384,6 +387,114 @@ func TestSplitObjectExpirationWithoutLink(t *testing.T) {
+	l, logBuf := testutil.NewBufferedLogger(t, zap.DebugLevel)
+	e := New(WithLogger(l))
+	_, err := e.AddShard(
+		shard.WithLogger(l),
+		shard.WithBlobstor(
+			newStorage(filepath.Join(dir, "fstree")),
+		),
+		shard.WithMetaBaseOptions(
+			meta.WithLogger(l),
+			meta.WithPath(filepath.Join(dir, "metabase")),
+			meta.WithPermissions(0700),
+			meta.WithEpochState(es),
+		),
+		shard.WithExpiredObjectsCallback(e.processExpiredObjects),
+		shard.WithGCRemoverSleepInterval(100*time.Millisecond),
+	)
+	require.NoError(t, err)
+
+	require.NoError(t, e.Open())
+	require.NoError(t, e.Init())
+	t.Cleanup(func() { _ = e.Close() })
+
+	cnr := cidtest.ID()
+	parentID := oidtest.ID()
+
+	parent := generateObjectWithCID(cnr)
+	parent.SetID(parentID)
+	parent.SetPayload(nil)
+	addExpirationAttribute(parent, es.CurrentEpoch())
+	parentAddr := object.AddressOf(parent)
+
+	const childCount = 3
+	children := make([]*objectSDK.Object, childCount)
+	childIDs := make([]oid.ID, childCount)
+
+	var firstID oid.ID
+	for i := range children {
+		children[i] = generateObjectWithCID(cnr)
+		if i == 0 {
+			firstID = children[i].GetID()
+		}
+		if i != 0 {
+			children[i].SetPreviousID(childIDs[i-1])
+			children[i].SetFirstID(firstID)
+		}
+		if i == len(children)-1 {
+			children[i].SetParent(parent)
+		}
+		children[i].SetPayload([]byte{byte(i), byte(i + 1), byte(i + 2)})
+		children[i].SetPayloadSize(3)
+		childIDs[i] = children[i].GetID()
+	}
+
+	var linkObj objectSDK.Object
+	linkObj.WriteLink(objectSDK.Link{})
+	linkObj.SetContainerID(cnr)
+	linkObj.SetParent(parent)
+	linkObj.SetParentID(parentID)
+	linkObj.SetFirstID(firstID)
+	linkObj.SetOwner(usertest.ID())
+	linkObj.CalculateAndSetPayloadChecksum()
+	require.NoError(t, linkObj.CalculateAndSetID())
+
+	linkAddr := object.AddressOf(&linkObj)
+
+	// Put link object and all children
+	require.NoError(t, e.Put(&linkObj, nil))
+	for i := range children {
+		require.NoError(t, e.Put(children[i], nil))
+	}
+
+	_, err = e.Get(linkAddr)
+	require.NoError(t, err)
+
+	_, err = e.Get(parentAddr)
+	var splitErr *objectSDK.SplitInfoError
+	require.ErrorAs(t, err, &splitErr)
+	require.Equal(t, linkObj.GetID(), splitErr.SplitInfo().GetLink())
+
+	// Now delete the link object to simulate missing link scenario
+	tomb := generateObjectWithCID(cnr)
+	tomb.SetType(objectSDK.TypeTombstone)
+	tombAddr := object.AddressOf(tomb)
+	require.NoError(t, e.Inhume(tombAddr, 1, linkAddr))
+
+	_, err = e.Get(linkAddr)
+	require.ErrorIs(t, err, statusSDK.ErrObjectAlreadyRemoved)
+
+	tickEpoch(es, e)
+
+	require.Eventually(t, func() bool {
+		// Check this way because Get returns ErrObjectNotFound for expired error
+		for _, sh := range e.sortedShards(parentAddr) {
+			_, err = sh.Get(parentAddr, false)
+			if errors.Is(err, statusSDK.ErrObjectNotFound) {
+				return true
+			}
+		}
+		return false
+	}, 1*time.Second, 100*time.Millisecond)
+
+	logBuf.AssertContainsMsg(zapcore.DebugLevel, "inhuming root object but no link object is found")
```

Comment:
```suggestion
	logBuf.AssertContainsMsg(zap.DebugLevel, "inhuming root object but no link object is found")
```
and below so `zapcore` pkg wont be needed